pages/swap/_swap.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <div class="h-full w-full flex flex-wrap py-4">
+    <VHocLoading :key="$route.fullPath" :status="status">
+    <div class="container">
+      <div class="mx-auto w-full sm:w-md">
+        <div class="flex flex-col mt-6">
+          <v-swap :set-market="setMarket" />
+        </div>
+      </div>
+    </div>
+    </VHocLoading>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Status, StatusType } from '@injectivelabs/utils'
+import { UiSpotMarketWithToken } from '@injectivelabs/ui-common'
+import VSwap from '~/components/partials/swap/index.vue'
+import { Modal } from '~/types'
+import { ORDERBOOK_POLLING_ENABLED } from '~/app/utils/constants'
+import { betaMarketSlugs } from '~/app/data/market'
+
+export default Vue.extend({
+  components: {
+    VSwap
+  },
+
+  data() {
+    return {
+      status: new Status(StatusType.Loading),
+      interval: 0 as any
+    }
+  },
+
+  computed: {
+    market(): UiSpotMarketWithToken | undefined {
+      return this.$accessor.spot.market
+    },
+
+    marketIsBeta(): boolean {
+      const { params } = this.$route
+
+      return betaMarketSlugs.includes(params.spot)
+    },
+
+    markets(): UiSpotMarketWithToken[] {
+      return this.$store.state.spot.markets
+    }
+  },
+
+  watch: {
```

Comment:
let's move the `getMarketFromRoute` logic to the child component so we can ensure that spot markets are fetched on the child component mounted lifecycle, instead of using `watch` here.